test/functional/tests/cache_ops/test_attach_detach.py

```diff
@@ -0,0 +1,258 @@
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrizex("cache_mode", [CacheMode.WB, CacheMode.WT])
+def test_attach_detach_md5sum(cache_mode):
+    """
+    title: Test for md5sum of file after attach/detach operation.
+    description: |
+        Test data integrity after detach/attach operations
+    pass_criteria:
+      - CAS doesn't crash during attach and detach.
+      - md5sums before and after operations match each other
+    """
+
+    with TestRun.step("Start cache and add core"):
+        cache_dev = TestRun.disks["cache"]
+        cache_dev.create_partitions([Size(2, Unit.GibiByte)])
+        cache_dev = cache_dev.partitions[0]
+
+        cache_dev2 = TestRun.disks["cache2"]
+        cache_dev2.create_partitions([Size(3, Unit.GibiByte)])
+        cache_dev2 = cache_dev2.partitions[0]
+
+        core_dev = TestRun.disks["core"]
+        core_dev.create_partitions([Size(6, Unit.GibiByte)])
+        core_dev = core_dev.partitions[0]
+
+        cache = casadm.start_cache(cache_dev, force=True, cache_mode=cache_mode)
+        core = cache.add_core(core_dev)
+
+    with TestRun.step(f"Change cache mode to {cache_mode}"):
+        cache.set_cache_mode(cache_mode)
+
+    with TestRun.step("Create a filesystem on the core device and mount it"):
+        TestRun.executor.run(f"rm -rf {mountpoint }")
+        create_directory(path=mountpoint)
+        core.create_filesystem(Filesystem.xfs)
+        core.mount(mountpoint)
+
+    with TestRun.step("Write data to the exported object"):
+        test_file_main = create_random_test_file(
+            target_file_path=posixpath.join(mountpoint, "test_file"),
+            file_size=Size(5, Unit.GibiByte),
+        )
+
+    with TestRun.step("Calculate test file md5sums before detach"):
+        test_file_md5sum_before = test_file_main.md5sum()
+
+    with TestRun.step("Detach cache device"):
+        cache.detach()
+
+    with TestRun.step("Attach different cache device"):
+        cache.attach(device=cache_dev2, force=True)
+
+    with TestRun.step("Calculate cache test file md5sums after cache attach"):
+        test_file_md5sum_after = test_file_main.md5sum()
+
+    with TestRun.step("Compare test file md5sums"):
+        if test_file_md5sum_before != test_file_md5sum_after:
+            TestRun.fail(
+                f"MD5 sums of core before and after do not match."
+                f"Expected: {test_file_md5sum_before}"
+                f"Actual: {test_file_md5sum_after}"
+            )
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrizex("cache_mode", CacheMode)
+def test_stop_cache_during_attach(cache_mode):
+    """
+    title: Test cache stop during attach.
+    description: |
+        Test the ability of OpenCAS to handle concurrent attach and cache stop.
```

Comment:
I'm not a fan of putting 'OpenCAS' in descriptions - we know what we're testing.
'Test for handling concurrent cache attach and stop' is sufficient.